Fix missing config/install info

bugfolder opened the pull request:
Several related changes, all to do with fixing the missing `return $prefixes;` in implementations of `hook_config_info()` and missing `$config->save();` in implementations of `hook_update_1000()`.

Along the way: code formatting (remove whitespace at ends of lines) and fixing an issue in grammar_parser that issued an error if `$data` was a plain string.

bugfolder committed:
Fix missing config/install info

bugfolder committed:
Make PHP 5.3 happy

Sorry, no ?? operator in PHP 5.3.

bugfolder commented:
The check on PHP 5.3 seems to be failing even before getting to the code:

https://travis-ci.com/github/backdrop-contrib/coder_upgrade/jobs/484900072

docwilmot commented on coder_upgrade.module:
```diff
@@ -183,5 +184,5 @@ function coder_upgrade_build_config_function($node, $reader) {
     'type' => T_WHITESPACE,
     'value' => 1,
   );
-  $container->insertBefore($node, $whitespace, 'whitespace');
```
Need to delete the array defnition just above.

bugfolder replied:
Thx, will do.

docwilmot commented on coder_upgrade.module:
```diff
@@ -162,16 +167,12 @@ function coder_upgrade_build_config_function($node, $reader) {
 
   $strings[] = '$prefixes[\'' . $_coder_upgrade_module_name . '.settings\'] = array(';
-  $strings[] = "  'label' => t('Module name settings'),";
+  $strings[] = "  'label' => t('<Module name> settings'),";
```
Is it not possible to get the actual module name and replace there?

bugfolder replied:
I didn't see where it would be found. Pointer would be welcome! I just added the angle brackets to make it clear that this was something that needed to be manually replaced (that wasn't clear to me the first few times I coder_upgraded a module).

bugfolder replied:
Found it! Latest version of the PR puts the actual human module name in the `hook_config_info()` implementation.

docwilmot commented on grammar_parser/list.inc:
```diff
@@ -2305,7 +2305,7 @@ class PGPString extends PGPExpression {
         case 'open':
         case 'close':
-          $string .= $data['value'];
+          $string .= isset($data['value']) ? $data['value'] : $data;
```
This doesnt seem related to the missing config/install issue, was it causing a problem? This is actually the Grammar Parser library code.

bugfolder replied:
It was causing a problem, but of wider scope: a warning about the unassigned variable. I think that when $data is a raw string, that's what should be appended.

bugfolder replied:
Should this change be broken out into a separate PR?

bugfolder replied:
On further consideration, yes, it should. There's a lot of other issues related to the grammar parser, and they should be addressed separately. So I've removed this change from the PR.

docwilmot reviewed (commented):
Just a few small questions. Thanks for the PR.

docwilmot commented:
The Travis integration isnt set up actually. Dont know why its now firing comments, it wasn't before.
Thanks for the PR.

bugfolder committed:
Remove superfluous $whitespace dfn.

bugfolder committed:
Add module name to hook_config_info()

Get the module name from the .info file and use it in the hook_config_info() implementation.

docwilmot commented on includes/main.inc:
```diff
@@ -559,12 +557,39 @@ function coder_upgrade_module_name($dirname, &$item) {
     $file_path = $path . $file;
     if (!is_dir($file_path)) {
-      if (in_array(pathinfo($file_path, PATHINFO_EXTENSION), $extensions)) {
+      $pathinfo = pathinfo($file_path, PATHINFO_EXTENSION);
```
minor nitpick: `$extension = pathinfo($file_path, PATHINFO_EXTENSION)` instead?

bugfolder replied:
Yup, done.

docwilmot commented on includes/main.inc:
```diff
@@ -559,12 +557,39 @@ function coder_upgrade_module_name($dirname, &$item) {
+/**
+ * Get the human name of the module from its .info file.
+ */
+function _coder_upgrade_module_human_name($file_path) {
```
Why not `backdrop_parse_info_file()`?

bugfolder replied:
Nice! Didn't know that existed. Done.

docwilmot reviewed (commented):
Couple more comments.

bugfolder committed:
Use backdrop_parse_info_file() to get human name

bugfolder committed:
Revert line 2308 change in list.inc.

Revert the change in line 2308 to its original value.

docwilmot commented:
Looks good; merged, thanks.
